From: NeilBrown <neilb@suse.com>
Subject: Fix kabi issues with new transport sharing code
References: bsc#1114893
Patch-mainline: Never, kabi

These structures might be viable by sub-modules, but aren't created by
them, so adding fields to the end is safe.

Signed-off-by: NeilBrown <neilb@suse.com>

---
 fs/nfs/internal.h         |    6 +++++-
 include/linux/nfs_fs_sb.h |    4 +++-
 2 files changed, 8 insertions(+), 2 deletions(-)

--- a/fs/nfs/internal.h
+++ b/fs/nfs/internal.h
@@ -81,7 +81,9 @@ struct nfs_client_initdata {
 	unsigned int nconnect;
 	struct net *net;
 	const struct rpc_timeout *timeparms;
+#ifndef __GENKSYMS__
 	unsigned int xprt_id;
+#endif
 };
 
 /*
@@ -103,7 +105,6 @@ struct nfs_parsed_mount_data {
 	unsigned int		minorversion;
 	char			*fscache_uniq;
 	bool			need_mount;
-	unsigned int		xprt_id;
 
 	struct {
 		struct sockaddr_storage	address;
@@ -126,6 +127,9 @@ struct nfs_parsed_mount_data {
 
 	struct security_mnt_opts lsm_opts;
 	struct net		*net;
+#ifndef __GENKSYMS__
+	unsigned int		xprt_id;
+#endif
 };
 
 /* mount_clnt.c */
--- a/include/linux/nfs_fs_sb.h
+++ b/include/linux/nfs_fs_sb.h
@@ -59,7 +59,6 @@ struct nfs_client {
 	unsigned int		cl_nconnect;	/* Number of connections */
 	struct rpc_cred		*cl_machine_cred;
 
-	unsigned int		cl_xprt_id;
 #if IS_ENABLED(CONFIG_NFS_V4)
 	struct list_head	cl_ds_clients; /* auth flavor data servers */
 	u64			cl_clientid;	/* constant */
@@ -121,6 +120,9 @@ struct nfs_client {
 #endif
 
 	struct net		*cl_net;
+#ifndef __GENKSYMS__
+	unsigned int		cl_xprt_id;
+#endif
 };
 
 /*
